engine/src/test/java/com/datatorrent/stram/client/StramClientUtilsTest.java

```diff
@@ -154,4 +160,209 @@ public InetSocketAddress getSocketAddr(String name, String defaultAddress, int d
     Assert.assertEquals(getHostString("192.168.1.2") + ":8032", StramClientUtils.getSocketConnectString(addresses.get(1)));
   }
 
+  /**
+   * apex.dfsRootDirectory not set: legacy behavior of getDTDFSRootDir()
+   * @throws IOException
+   *
+   */
+  @Test
+  public void test_getApexDFSRootDir_Legacy() throws IOException
+  {
+    Configuration conf = new YarnConfiguration(new Configuration(false));
+    conf.set(StramClientUtils.DT_DFS_ROOT_DIR, "/a/b/c");
+    conf.setBoolean(StramUserLogin.DT_APP_PATH_IMPERSONATED, false);
+
+    FileSystem fs = FileSystem.newInstance(conf);
+    Path path = StramClientUtils.getApexDFSRootDir(fs, conf);
+    Assert.assertEquals("file:/a/b/c", path.toString());
+  }
+
+  /**
+   * apex.dfsRootDirectory set: absolute path
+   * @throws IOException
+   *
+   */
+  @Test
+  public void test_getApexDFSRootDir_ApexAbsolutePath() throws IOException
+  {
+    Configuration conf = new YarnConfiguration(new Configuration(false));
+    conf.set(StramClientUtils.APEX_APP_DFS_ROOT_DIR, "/x/y/z");
+    conf.setBoolean(StramUserLogin.DT_APP_PATH_IMPERSONATED, false);
+
+    FileSystem fs = FileSystem.newInstance(conf);
+    UserGroupInformation testUser = UserGroupInformation.createUserForTesting("testUser1", new String[]{""});
+    UserGroupInformation.setLoginUser(testUser);
+    Path path = StramClientUtils.getApexDFSRootDir(fs, conf);
+    Assert.assertEquals("file:/x/y/z", path.toString());
+  }
+
+  /**
+   * apex.dfsRootDirectory set: absolute path with scheme
+   * @throws IOException
+   *
+   */
+  @Test
+  public void test_getApexDFSRootDir_ApexAbsolutePathWithScheme() throws IOException
+  {
+    Configuration conf = new YarnConfiguration(new Configuration(false));
+    conf.set(StramClientUtils.APEX_APP_DFS_ROOT_DIR, "file:/p/q/r");
+    conf.setBoolean(StramUserLogin.DT_APP_PATH_IMPERSONATED, false);
+
+    FileSystem fs = FileSystem.newInstance(conf);
+    UserGroupInformation testUser = UserGroupInformation.createUserForTesting("testUser1", new String[]{""});
+    UserGroupInformation.setLoginUser(testUser);
+    Path path = StramClientUtils.getApexDFSRootDir(fs, conf);
+    Assert.assertEquals("file:/p/q/r", path.toString());
+  }
+
+  /**
+   * apex.dfsRootDirectory set: absolute path with %USER_NAME%
+   * @throws IOException
+   * @throws InterruptedException
+   *
+   */
+  @Test
+  public void test_getApexDFSRootDir_ApexAbsolutePathWithUSERNAME() throws IOException, InterruptedException
```

Comment:
My personal preference is to have long and self-describing names for methods, identifiers etc but I will incorporate your comments.